### PR TITLE
Updating Search Schemas & Reindexing Data page for KV

### PR DESCRIPTION
**Pull request by erjohnson**
RFC

### CODE REVIEW

File: content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  riak_kv-2.1.4:
+    name: "Updating / Reindexing Search Schemas"
+    identifier: "cluster_operations_search_schemas"
+    weight: 117
+    parent: "managing_cluster_operations"
+toc: true
+canonical_link: "https://docs.basho.com/riak/kv/latest/using/cluster-operations/update-reindex-search-schema"
+---
+
+The following page provides instruction on updating [search schemas](../../../developing/usage/search-schemas) as well as reindexing existing data.
```
Comment (laurenrother) [inline content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md]: Comma after search schemas.

File: content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+## Update the Search Schema
+
+First, make updates to the schema file, which is typically found here `$RIAK_HOST/search/index/famous`.
```
Comment (laurenrother) [inline content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md]: 1. "make updates to" should be "update the schema file"
2. What kind of updates? Why?

File: content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+For example, the following creates or updates an index called `famous` that uses the default schema:
```
Comment (laurenrother) [inline content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md]: Wait, does the below code update an index that already uses the default schema or does it update the index TO use the default schema? If not the latter, then what is it updating/creating the index to do?

File: content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+```bash
+curl -XPUT $RIAK_HOST/search/index/famous \ -H 'Content-Type: application/json' \ -d '{"schema":"_yz_default"}'
+```
+
+Then update the new schema within Riak KV by either:
+
+* Performing a [rolling restart](../../repair-recovery/rolling-restart) to trigger a Solr reload of the schema.
+* Or starting a `riak attach` session and run:
```
Comment (laurenrother) [inline content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md]: *running

File: content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  `yz_index:reload(<<"index_name">>).`
+
+> **Note:**
+>
+> Please note the `riak attach` method used above is a blocking, cluster-wide call but it will trigger a live reload of the Solr schema. New data will now contain the updated fields but existing data will not have their indexes updated.
+
+## Reindex Existing Data
+
+The first option to reindex existing data is rePUT all of your data.  You can do this by using a script or the [data migrator tool](https://github.com/dankerrigan/riak-data-migrator).
```
Comment (laurenrother) [inline content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md]: 1. *re-PUT
2. What kind of script? Something they put together or something we recommend?

File: content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+The second option for reindexing existing data is to:
+
+1. Delete all of your Solr data from the index that you changed the schema on.  You do not need to delete the configurations.
```
Comment (laurenrother) [inline content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md]: Don't end a sentence on a preposition. "Delete all of your Solr data from the index where you changed the schema." or "..from the index on which you..."

File: content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+2. Expire the YZ AAE trees by entering a `riak attach` session and running the following command:  `rpc:multicall([node() | nodes()], yz_entropy_mgr, expire_trees, []).`
```
Comment (laurenrother) [inline content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md]: @TheOnlyRew @angrycub In your experience do people usually know how to user `riak attach`? Do we need to explain more?

File: content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+3. Allow the trees to rebuild with the new index.  You can monitor the progress of this activity with `riak-admin search aae-status`
```
Comment (laurenrother) [inline content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md]: What does a good progression look like? What does a bad one look like? What does a person do if they are seeing errors?

File: content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+
+## AAE Tuning
+
+By default search reindexing through AAE happens at the same rate as KV AAE. AAE trees are rebuilt at a rate of 1 tree per node per hour, and AAE trees are expired once every week. With these settings it likely will take a great deal of time for the index to rebuild after the trees expire.
```
Comment (laurenrother) [inline content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md]: Comma after default

File: content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+
+The [Active Anti Entropy][../active-anti-entropy] settings are described in our documentation. Of particular interest to you I think will be the following settings in `riak.conf` format, shown with their default values:
```
Comment (laurenrother) [inline content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md]: It's weird that AAE is spelled out here but not above when it appears for the first time. It should probably get called out above and (abbreviated).

"Of particular interest will be the following `riak.conf`settings (shown with default values):"

File: content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+
+```
+
+anti_entropy.concurrency_limit = 2
+anti_entropy.tree.build_limit.number = 1
+anti_entropy.tree.build_limit.per_timespan = 1h
+anti_entropy.tree.expiry = 1w
+```
+
+You can experiment with setting `anti_entropy.tree.build_limit.per_timespan` to something shorter, perhaps 15m and/or raising the `anti_entropy.tree.build_limit.number`. You can also raise the `anti_entropy.concurrency_limit` to, for example, 4. These changes should be made in `riak.conf` and do require a node restart to be applied. Please monitor the impact of these changes on your cluster's performance. We cannot currently adjust the AAE settings for KV and Search separately through `riak.conf`.  Changing these settings will allow the trees to rebuild faster and thus have your data reindexed faster; however, will place additional load on your cluster.
```
Comment (laurenrother) [inline content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md]: comma after 15m.

Cut ,for example, in front of 4. 

What does raising the limit to 4 do?

File: content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+
+You can monitor the progress of the re-indexing with the command Ian mentioned, `riak-admin search aae_status`. Please be aware that until all partitions have been reindexed you may see inconsistent results due to the way the query plan is accessing different Solr backends that may not yet have all the data reindexed.
```
Comment (laurenrother) [inline content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md]: Remove mention of Ian.

File: content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md (hunk 7 above)
Comment (angrycub) [inline content/riak/kv/2.1.4/using/cluster-operations/update-search-schema-reindex-data.md]: I would say that many people who are working at this level should have a passing familiarity with the command.   That said, I do wish we had a one page primer on running `riak attach`, the importance of full stops, and how to exit properly.   It's a doc I have been dreaming about for far too long.

**Pull request closed without merging**